src/OpenTracing/SpanReference.php

```diff
@@ -2,20 +2,93 @@
 
 namespace OpenTracing;
 
-interface SpanReference
+use OpenTracing\Exceptions\InvalidReferenceArgument;
+
+final class SpanReference
 {
+    const CHILD_OF = 'child_of';
+    const FOLLOWS_FROM = 'follows_from';
+
     /**
-     * @return bool
+     * @var SpanContext
      */
-    public function isTypeChildOf();
+    private $context;
 
     /**
-     * @return bool
+     * @var string
+     */
+    private $type;
+
+    private function __construct(SpanContext $context, $type)
+    {
+        $this->context = $context;
+        $this->type = $type;
+    }
+
+    /**
+     * @param SpanContext|Span $context
+     * @param string $type
+     * @throws InvalidReferenceArgument on empty type
+     * @return SpanReference when context is invalid
+     */
+    public static function create($context, $type)
+    {
+        if (empty($type)) {
+            throw InvalidReferenceArgument::emptyType();
```

Comment:
I don't understand why the constructor itself should allow this but this method  doesn't